File: garnet/bin/iquery/utils_test.cc
// Copyright 2018 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "garnet/bin/iquery/utils.h"

#include "gtest/gtest.h"

namespace {

TEST(IqueryUtils, IsStringPrintable) {
  EXPECT_TRUE(iquery::IsStringPrintable("hello"));
  EXPECT_TRUE(iquery::IsStringPrintable("hello world"));
  EXPECT_TRUE(iquery::IsStringPrintable("hello\tworld"));
  EXPECT_TRUE(iquery::IsStringPrintable("hello\nworld\r\nagain"));
  // <smiling face> :-) <filled black star>.
  // \x05 is not printable ASCII, so this checks that we properly increment the
  // index.
  EXPECT_TRUE(iquery::IsStringPrintable("\u263A :-) \u2605"));
  EXPECT_FALSE(iquery::IsStringPrintable("hello\x6"));
  EXPECT_FALSE(iquery::IsStringPrintable("hello\x80"));
  EXPECT_FALSE(iquery::IsStringPrintable(std::string("hello\0", 6)));
}

}  // namespace

File: tools/fuzz/utils_test.go
// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package fuzz

import (
	"io"
	"strings"
	"testing"
)

func TestParallelMultiReader(t *testing.T) {
	r1, w1 := io.Pipe()
	r2, w2 := io.Pipe()
	line1 := "hello\n"
	line2 := "world\n"

	combined := parallelMultiReader(r1, r2)
	io.WriteString(w1, line1)
	io.WriteString(w2, line2)
	w1.Close()
	w2.Close()

	out, err := io.ReadAll(combined)
	if err != nil {
		t.Fatalf("error reading: %s", err)
	}

	if !strings.Contains(string(out), line1) || !strings.Contains(string(out), line2) {
		t.Fatalf("unexpected output: %s", out)
	}
}
